ndk/src/font.rs

```diff
@@ -0,0 +1,401 @@
+            FontWeight::BOLD => writeln!(f, "Bold"),
+            FontWeight::EXTRA_BOLD => writeln!(f, "Extra Bold (Ultra Bold)"),
+            FontWeight::BLACK => writeln!(f, "Black (Heavy)"),
+            _ => writeln!(f, "{}", self.0),
+        }
+    }
+}
+
+/// The error type returned when an invalie font weight value is passed.
+#[derive(Debug)]
+pub struct TryFromU16Error(());
+
+impl fmt::Display for TryFromU16Error {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            fmt,
+            "font weight must be positive and less than or equal to 1000"
+        )
+    }
+}
+
+impl std::error::Error for TryFromU16Error {}
+
+impl TryFrom<u16> for FontWeight {
+    type Error = TryFromU16Error;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        FontWeight::new(value)
+    }
+}
+
+/// A 4-byte integer representing an OpenType axis tag.
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct AxisTag(u32);
+
+impl AxisTag {
+    pub const fn from_be(value: u32) -> std::result::Result<Self, TryFromU32Error> {
+        // Each byte in a tag must be in the range 0x20 to 0x7E.
+        // See https://learn.microsoft.com/en-us/typography/opentype/spec/otff#data-types for details.
+        macro_rules! check_in_valid_range {
+            ($($byte: expr),+) => {
+                $(
+                    if !(0x20 <= ($byte) && ($byte) <= 0x7E) {
+                        return Err(TryFromU32Error(()));
+                    }
+                )+
+            };
+        }
+
+        let bytes = value.to_be_bytes();
+        check_in_valid_range!(bytes[0], bytes[1], bytes[2], bytes[3]);
+
+        Ok(Self(value))
+    }
+}
+
+impl fmt::Display for AxisTag {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let bytes = self.0.to_be_bytes();
+        write!(
+            f,
+            "{}{}{}{}",
+            bytes[0] as char, bytes[1] as char, bytes[2] as char, bytes[3] as char
+        )
+    }
+}
+
+impl fmt::Debug for AxisTag {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "AxisTag(")?;
+        fmt::Display::fmt(self, f)?;
+        write!(f, " {:#x})", self.0)
+    }
+}
+
+/// The error type returned when an invalie font weight value is passed.
+#[derive(Debug)]
+pub struct TryFromU32Error(());
+
+impl fmt::Display for TryFromU32Error {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            fmt,
+            "each byte in an axis tag must be in the range 0x20 to 0x7E"
+        )
+    }
+}
+
+impl std::error::Error for TryFromU32Error {}
+
+/// A native [`AFont *`]
+///
+/// [`AFont *`]: https://developer.android.com/ndk/reference/group/font#afont_close
+#[cfg(feature = "api-level-29")]
+#[derive(Debug)]
+pub struct Font {
+    ptr: NonNull<ffi::AFont>,
+}
+
+#[cfg(feature = "api-level-29")]
+impl Font {
+    /// Create an `Font` from a pointer
```

Comment:
```suggestion
    /// Create a `Font` from a pointer
```

Reply:
And this should mention that it assumes ownership of `ptr` (since you drop `_close` it later). `NativeWindow` docs are probably the most up-to-date as a reference for how the docs should look.

Reply:
Fixed!